### PR TITLE
Handle PBIX files wtihout internal datasets more gracefully

### LINKED ISSUE #27: Unable to refresh state if PBIX file contains no dataset
I have no idea how it is possible, but it seems that you can actually create PBIX files that do not create a dataset. That means, that `datatset_id` is empty in these cases. While it isn't much of an issue for initial creation (this attribute is just null in the state file), it is a  problem for succeeding updates to the infrastructure, as the refreshing step fails with the following error:

```
Error: Get "https://api.powerbi.com/v1.0/myorg/datasets//parameters": status code '404 Not Found' with body {"error":{"code":"ItemNotFound","message":"Dataset parameters is not found."}}
```

The error is kind of obvious, as the URL is build of the parameter `datatset_id` which is empty in this case.

I have no idea how you can create such PBIX files (and I cannot share the one I have right now because of intellectual property), but I will try to debug it further and let you know.

### PR DESCRIPTION
**Pull request by alex-davies**
Prevent trying to fetch datasources and parameters if there are no datasets. Return error if trying to set parameters or datasets on pbixs that do not have datasets

Closes #27

### CODE REVIEW

File: internal/powerbi/resource_pbix.go [hunk 1]
```diff
@@ -303,11 +304,16 @@ func setPBIXParameters(d *schema.ResourceData, meta interface{}) error {
...
 	parameter := d.Get("parameter").(*schema.Set)
-	datasetID := d.Get("dataset_id").(string)
+	datasetID, datasetOk := d.GetOk("dataset_id")
 	groupID := d.Get("workspace_id").(string)
 	if parameter != nil {
 		parameterList := parameter.List()
 		if len(parameterList) > 0 {
+
+			if !datasetOk {
+				return fmt.Errorf("Unable to update parameters on a PBIX file that does not contain a dataset")
```
Comment (alxy) [inline internal/powerbi/resource_pbix.go:314]: Not sure if this should really be considered an "error", as technically it is completely fine to have pbix resources without datasets.

Comment (alex-davies) [reply internal/powerbi/resource_pbix.go:314]: That code is for when updating parameters. This error occurs in the case if you have a PBIX without a dataset and try to set parameters (or datasources, which has similar code) in the terraform

```
resource "powerbi_workspace" "test" {
	name = "My Worksapce"
}

resource "powerbi_pbix" "test" {
	workspace_id = "${powerbi_workspace.test.id}"
	name = "PBIX without a dataset"
	source = "./pbix_wihtout_dataset.pbix"
	source_hash = "filemd5(./pbix_wihtout_dataset.pbix)"

# Setting a parameter - but as parameters are set at the dataset this doesnt make any sense if the pbix doesnt have a dataset
	parameter {
		name = "ParamOne"
		value = "NewParamValueOne"
	}
}
```

I could ignore the parameters for cases of datasetless pbix, but if you have this case something is probably happening that you are not expecting, so think its safer to throw an error

Comment (alxy) [reply internal/powerbi/resource_pbix.go:314]: Makes sense!

**Comment by alxy** [comment]
Hi @alex-davies I see that all tests pass now. Would you mind merging this and releasing a new version?

**Comment by alex-davies** [comment]
Although the test are passing, I'm cheating in order to get them to do so

a report-only PBIX needs the dataset to exist and needs access to that dataset. My test case PBIX has a dataset that exists only because I manually added it and gave it the appropriate permissions. On a blank PowerBI workspace the tests will fail. And the only way to fix it would be to modify the PBIX file to point to a different dataset that does exist and has permissions to.

However I think i've found a way to hack a pbix to change the dataset ID it is using, although I wouldn't be comfortable yet doing this as part of the main provider, I think it is fine to do it as part of the tests to make them less environment specific

**Comment by alxy** [comment]
@alex-davies Can you release this as a new version to the terraform  registry? It would help us very much! Many thanks in advance :)

**Comment by alex-davies** [comment]
@alxy apologies I thought had I had created release, but it was stuck in draft. Release has been created and is now available from the terraform registry

**Comment by alxy** [comment]
Thank you!

**Pull request merged**